src/components/PicturesListComponent.js

```diff
@@ -14,17 +25,23 @@ const PicturesListComponent = (props) => {
         loadInBrowser,
         handleSort,
         loadModal,
+        isGridEnabled,
     } = props;
+    const imageResolution = getImageResolution();
 
     return (
         <>
             {error && <ErrorElement message={error} />}
 
             <FlatList
                 data={data}
-                contentContainerStyle={styles.listContainer}
+                key={isGridEnabled}
+                numColumns={isGridEnabled ? 3 : 1}
+                contentContainerStyle={determineFlatListStyle(isGridEnabled)}
                 renderItem={({ item }) => (
```

Comment:
Create a function for renderItem - useCallback.

Every render rerender all elemetns curently